packages/apollo-server-core/src/__tests__/ApolloServerBase.test.ts

```diff
@@ -48,3 +48,42 @@ describe('ApolloServerBase construction', () => {
     );
   });
 });
+
+describe('environment variables', () => {
+  const OLD_ENV = process.env;
+
+  beforeEach(() => {
+    jest.resetModules();
+    process.env = { ...OLD_ENV };
+    delete process.env.ENGINE_API_KEY;
+    delete process.env.APOLLO_KEY;
+  });
+
+  afterEach(() => {
+    process.env = OLD_ENV;
+  });
+
+  it('constructs a reporting agent with the legacy env var and warns', async () => {
```

Comment:
Perhaps _in addition_ to server, but I specifically wanted to capture the "only warn once" when Apollo Server is created, since there are multiple calls to the warn-spewing reporting agent code.